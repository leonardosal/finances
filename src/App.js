import React from 'react';

import Home from './containers/Home';
import Header from './components/Header';

import './styles/global.scss';

const App = () => (
  <>
    <Header />
    <Home />
  </>
);

export default App;
