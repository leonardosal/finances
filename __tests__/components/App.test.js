import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from '../../src/App';

it('should render App component', () => {
  const component = mount(<App />);
  expect(toJson(component)).toMatchSnapshot();
});
