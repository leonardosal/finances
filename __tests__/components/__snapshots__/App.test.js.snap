// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component 1`] = `
<App>
  <Header>
    <header
      className="header"
    >
      <span>
        Finances
      </span>
    </header>
  </Header>
  <Home>
    <div
      className="container"
    >
      <div
        className="box"
      >
        <CardInfo
          balance={0}
          handleClick={[Function]}
        >
          <div
            className="card-info"
          >
            <div>
              <h5>
                Saldo
              </h5>
              <h2>
                R$0,00
              </h2>
            </div>
            <Button
              id="btn-open"
              label="Adicionar Transação"
              onClick={[Function]}
              type="button"
            >
              <button
                className="button"
                id="btn-open"
                onClick={[Function]}
                type="button"
              >
                Adicionar Transação
              </button>
            </Button>
          </div>
        </CardInfo>
      </div>
      <div
        className="separator"
      />
      <div
        className="box"
      >
        <Table
          transactions={Array []}
        />
      </div>
    </div>
    <Modal
      onChangeField={[Function]}
      show={false}
      submitForm={[Function]}
      toggleModal={[Function]}
      transaction={
        Object {
          "description": "",
          "type": "DEBIT",
          "value": "",
        }
      }
    >
      <div
        className="modal"
        id="modal"
      >
        <div
          className="modal-content"
        >
          <h3>
            Adicionar transação
          </h3>
          <form
            className="form"
            id="transactionForm"
            name="transactionForm"
            onSubmit={[Function]}
          >
            <div
              className="field-container"
            >
              <span>
                Tipo
              </span>
              <select
                id="type"
                name="type"
                onChange={[Function]}
                value="DEBIT"
              >
                <option
                  value="DEBIT"
                >
                  Despesa
                </option>
                <option
                  value="CREDIT"
                >
                  Receita
                </option>
              </select>
            </div>
            <div
              className="field-container"
            >
              <span>
                Valor
              </span>
              <CurrencyInput
                allowNegative={false}
                autoFocus={false}
                decimalSeparator=","
                id="value"
                inputType="text"
                name="value"
                onChange={[Function]}
                onChangeEvent={[Function]}
                precision="2"
                prefix=""
                required={true}
                selectAllOnFocus={false}
                suffix=""
                thousandSeparator="."
                value=""
              >
                <input
                  id="value"
                  name="value"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onMouseUp={[Function]}
                  required={true}
                  type="text"
                  value="0,00"
                />
              </CurrencyInput>
            </div>
            <div
              className="field-container"
            >
              <span>
                Descrição
              </span>
              <input
                id="description"
                name="description"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="footer"
            >
              <Button
                id="btn-save"
                label="Salvar"
                onClick={[Function]}
                type="submit"
              >
                <button
                  className="button"
                  id="btn-save"
                  onClick={[Function]}
                  type="submit"
                >
                  Salvar
                </button>
              </Button>
              <Button
                id="btn-cancel"
                label="Cancelar"
                onClick={[Function]}
                type="button"
              >
                <button
                  className="button"
                  id="btn-cancel"
                  onClick={[Function]}
                  type="button"
                >
                  Cancelar
                </button>
              </Button>
            </div>
          </form>
        </div>
      </div>
    </Modal>
  </Home>
</App>
`;
